File: managed/src/SwiftlyS2.Generated/Natives/CEntityKeyValues.cs
#pragma warning disable CS0649
#pragma warning disable CS0169

using System.Buffers;
using System.Text;
using System.Threading;
using SwiftlyS2.Shared.Natives;

namespace SwiftlyS2.Core.Natives;

internal static class NativeCEntityKeyValues {

  private unsafe static delegate* unmanaged<nint> _Allocate;

  public unsafe static nint Allocate() {
    var ret = _Allocate();
    return ret;
  }

  private unsafe static delegate* unmanaged<nint, void> _Deallocate;

  public unsafe static void Deallocate(nint keyvalues) {
    _Deallocate(keyvalues);
  }

  private unsafe static delegate* unmanaged<nint, byte*, byte> _GetBool;

  public unsafe static bool GetBool(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetBool(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret == 1;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, int> _GetInt;

  public unsafe static int GetInt(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetInt(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, uint> _GetUint;

  public unsafe static uint GetUint(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetUint(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, long> _GetInt64;

  public unsafe static long GetInt64(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetInt64(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, ulong> _GetUint64;

  public unsafe static ulong GetUint64(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetUint64(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, float> _GetFloat;

  public unsafe static float GetFloat(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetFloat(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, double> _GetDouble;

  public unsafe static double GetDouble(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetDouble(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<byte*, nint, byte*, int> _GetString;

  public unsafe static string GetString(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetString(null, keyvalues, keyBufferPtr);
      var retBuffer = pool.Rent(ret + 1);
      fixed (byte* retBufferPtr = retBuffer) {
        ret = _GetString(retBufferPtr, keyvalues, keyBufferPtr);
        var retString = Encoding.UTF8.GetString(retBufferPtr, ret);
        pool.Return(retBuffer);
        pool.Return(keyBuffer);
        return retString;
      }
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, nint> _GetPtr;

  public unsafe static nint GetPtr(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetPtr(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, CUtlStringToken> _GetStringToken;

  public unsafe static CUtlStringToken GetStringToken(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetStringToken(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, Color> _GetColor;

  public unsafe static Color GetColor(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetColor(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, Vector> _GetVector;

  public unsafe static Vector GetVector(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetVector(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, Vector2D> _GetVector2D;

  public unsafe static Vector2D GetVector2D(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetVector2D(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, Vector4D> _GetVector4D;

  public unsafe static Vector4D GetVector4D(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetVector4D(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, QAngle> _GetQAngle;

  public unsafe static QAngle GetQAngle(nint keyvalues, string key) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      var ret = _GetQAngle(keyvalues, keyBufferPtr);
      pool.Return(keyBuffer);
      return ret;
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, byte, void> _SetBool;

  public unsafe static void SetBool(nint keyvalues, string key, bool value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetBool(keyvalues, keyBufferPtr, value ? (byte)1 : (byte)0);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, int, void> _SetInt;

  public unsafe static void SetInt(nint keyvalues, string key, int value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetInt(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, uint, void> _SetUint;

  public unsafe static void SetUint(nint keyvalues, string key, uint value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetUint(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, long, void> _SetInt64;

  public unsafe static void SetInt64(nint keyvalues, string key, long value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetInt64(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, ulong, void> _SetUint64;

  public unsafe static void SetUint64(nint keyvalues, string key, ulong value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetUint64(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, float, void> _SetFloat;

  public unsafe static void SetFloat(nint keyvalues, string key, float value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetFloat(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, double, void> _SetDouble;

  public unsafe static void SetDouble(nint keyvalues, string key, double value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetDouble(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, byte*, void> _SetString;

  public unsafe static void SetString(nint keyvalues, string key, string value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    var valueLength = Encoding.UTF8.GetByteCount(value);
    var valueBuffer = pool.Rent(valueLength + 1);
    Encoding.UTF8.GetBytes(value, valueBuffer);
    valueBuffer[valueLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      fixed (byte* valueBufferPtr = valueBuffer) {
        _SetString(keyvalues, keyBufferPtr, valueBufferPtr);
        pool.Return(keyBuffer);
        pool.Return(valueBuffer);
      }
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, nint, void> _SetPtr;

  public unsafe static void SetPtr(nint keyvalues, string key, nint value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetPtr(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, CUtlStringToken, void> _SetStringToken;

  public unsafe static void SetStringToken(nint keyvalues, string key, CUtlStringToken value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetStringToken(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, Color, void> _SetColor;

  public unsafe static void SetColor(nint keyvalues, string key, Color value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetColor(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, Vector, void> _SetVector;

  public unsafe static void SetVector(nint keyvalues, string key, Vector value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetVector(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, Vector2D, void> _SetVector2D;

  public unsafe static void SetVector2D(nint keyvalues, string key, Vector2D value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetVector2D(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, Vector4D, void> _SetVector4D;

  public unsafe static void SetVector4D(nint keyvalues, string key, Vector4D value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetVector4D(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }

  private unsafe static delegate* unmanaged<nint, byte*, QAngle, void> _SetQAngle;

  public unsafe static void SetQAngle(nint keyvalues, string key, QAngle value) {
    var pool = ArrayPool<byte>.Shared;
    var keyLength = Encoding.UTF8.GetByteCount(key);
    var keyBuffer = pool.Rent(keyLength + 1);
    Encoding.UTF8.GetBytes(key, keyBuffer);
    keyBuffer[keyLength] = 0;
    fixed (byte* keyBufferPtr = keyBuffer) {
      _SetQAngle(keyvalues, keyBufferPtr, value);
      pool.Return(keyBuffer);
    }
  }
}

File: src/scripting/sdk/centitykeyvalues.cpp
/************************************************************************************************
 * SwiftlyS2 is a scripting framework for Source2-based games.
 * Copyright (C) 2023-2026 Swiftly Solution SRL via Sava Andrei-Sebastian and it's contributors
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 ************************************************************************************************/

#include <api/interfaces/manager.h>
#include <scripting/scripting.h>

#include <public/entity2/entitykeyvalues.h>
#include <public/tier1/utlstringtoken.h>

void* Bridge_CEntityKeyValues_Allocate()
{
    CEntityKeyValues* kv = new CEntityKeyValues();
    kv->AddRef();
    return kv;
}

void Bridge_CEntityKeyValues_Deallocate(void* ptr)
{
    ((CEntityKeyValues*)ptr)->Release();
}

bool Bridge_CEntityKeyValues_GetBool(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetBool(keyName);
}

int Bridge_CEntityKeyValues_GetInt(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetInt(keyName);
}

uint32_t Bridge_CEntityKeyValues_GetUint(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetUint(keyName);
}

int64_t Bridge_CEntityKeyValues_GetInt64(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetInt64(keyName);
}

uint64_t Bridge_CEntityKeyValues_GetUint64(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetUint64(keyName);
}

float Bridge_CEntityKeyValues_GetFloat(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetFloat(keyName);
}

double Bridge_CEntityKeyValues_GetDouble(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetDouble(keyName);
}

int Bridge_CEntityKeyValues_GetString(char* out, void* keyvalues, const char* keyName)
{
    static std::string s;
    s = ((CEntityKeyValues*)keyvalues)->GetString(keyName);

    if (out) strcpy(out, s.c_str());

    return s.size();
}

void* Bridge_CEntityKeyValues_GetPtr(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetPtr(keyName);
}

CUtlStringToken Bridge_CEntityKeyValues_GetStringToken(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetStringToken(keyName);
}

Color Bridge_CEntityKeyValues_GetColor(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetColor(keyName);
}

Vector Bridge_CEntityKeyValues_GetVector(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetVector(keyName);
}

Vector2D Bridge_CEntityKeyValues_GetVector2D(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetVector2D(keyName);
}

Vector4D Bridge_CEntityKeyValues_GetVector4D(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetVector4D(keyName);
}

QAngle Bridge_CEntityKeyValues_GetQAngle(void* keyvalues, const char* keyName)
{
    return ((CEntityKeyValues*)keyvalues)->GetQAngle(keyName);
}

void Bridge_CEntityKeyValues_SetBool(void* keyvalues, const char* keyName, bool value)
{
    ((CEntityKeyValues*)keyvalues)->SetBool(keyName, value);
}

void Bridge_CEntityKeyValues_SetInt(void* keyvalues, const char* keyName, int value)
{
    ((CEntityKeyValues*)keyvalues)->SetInt(keyName, value);
}

void Bridge_CEntityKeyValues_SetUint(void* keyvalues, const char* keyName, uint32_t value)
{
    ((CEntityKeyValues*)keyvalues)->SetUint(keyName, value);
}

void Bridge_CEntityKeyValues_SetInt64(void* keyvalues, const char* keyName, int64_t value)
{
    ((CEntityKeyValues*)keyvalues)->SetInt64(keyName, value);
}

void Bridge_CEntityKeyValues_SetUint64(void* keyvalues, const char* keyName, uint64_t value)
{
    ((CEntityKeyValues*)keyvalues)->SetUint64(keyName, value);
}

void Bridge_CEntityKeyValues_SetFloat(void* keyvalues, const char* keyName, float value)
{
    ((CEntityKeyValues*)keyvalues)->SetFloat(keyName, value);
}

void Bridge_CEntityKeyValues_SetDouble(void* keyvalues, const char* keyName, double value)
{
    ((CEntityKeyValues*)keyvalues)->SetDouble(keyName, value);
}

void Bridge_CEntityKeyValues_SetString(void* keyvalues, const char* keyName, const char* value)
{
    ((CEntityKeyValues*)keyvalues)->SetString(keyName, value);
}

void Bridge_CEntityKeyValues_SetPtr(void* keyvalues, const char* keyName, void* value)
{
    ((CEntityKeyValues*)keyvalues)->SetPtr(keyName, value);
}

void Bridge_CEntityKeyValues_SetStringToken(void* keyvalues, const char* keyName, CUtlStringToken value)
{
    ((CEntityKeyValues*)keyvalues)->SetStringToken(keyName, value);
}

void Bridge_CEntityKeyValues_SetColor(void* keyvalues, const char* keyName, Color value)
{
    ((CEntityKeyValues*)keyvalues)->SetColor(keyName, value);
}

void Bridge_CEntityKeyValues_SetVector(void* keyvalues, const char* keyName, Vector value)
{
    ((CEntityKeyValues*)keyvalues)->SetVector(keyName, value);
}

void Bridge_CEntityKeyValues_SetVector2D(void* keyvalues, const char* keyName, Vector2D value)
{
    ((CEntityKeyValues*)keyvalues)->SetVector2D(keyName, value);
}

void Bridge_CEntityKeyValues_SetVector4D(void* keyvalues, const char* keyName, Vector4D value)
{
    ((CEntityKeyValues*)keyvalues)->SetVector4D(keyName, value);
}

void Bridge_CEntityKeyValues_SetQAngle(void* keyvalues, const char* keyName, QAngle value)
{
    ((CEntityKeyValues*)keyvalues)->SetQAngle(keyName, value);
}

DEFINE_NATIVE("CEntityKeyValues.Allocate", Bridge_CEntityKeyValues_Allocate);
DEFINE_NATIVE("CEntityKeyValues.Deallocate", Bridge_CEntityKeyValues_Deallocate);
DEFINE_NATIVE("CEntityKeyValues.GetBool", Bridge_CEntityKeyValues_GetBool);
DEFINE_NATIVE("CEntityKeyValues.GetInt", Bridge_CEntityKeyValues_GetInt);
DEFINE_NATIVE("CEntityKeyValues.GetUint", Bridge_CEntityKeyValues_GetUint);
DEFINE_NATIVE("CEntityKeyValues.GetInt64", Bridge_CEntityKeyValues_GetInt64);
DEFINE_NATIVE("CEntityKeyValues.GetUint64", Bridge_CEntityKeyValues_GetUint64);
DEFINE_NATIVE("CEntityKeyValues.GetFloat", Bridge_CEntityKeyValues_GetFloat);
DEFINE_NATIVE("CEntityKeyValues.GetDouble", Bridge_CEntityKeyValues_GetDouble);
DEFINE_NATIVE("CEntityKeyValues.GetString", Bridge_CEntityKeyValues_GetString);
DEFINE_NATIVE("CEntityKeyValues.GetPtr", Bridge_CEntityKeyValues_GetPtr);
DEFINE_NATIVE("CEntityKeyValues.GetStringToken", Bridge_CEntityKeyValues_GetStringToken);
DEFINE_NATIVE("CEntityKeyValues.GetColor", Bridge_CEntityKeyValues_GetColor);
DEFINE_NATIVE("CEntityKeyValues.GetVector", Bridge_CEntityKeyValues_GetVector);
DEFINE_NATIVE("CEntityKeyValues.GetVector2D", Bridge_CEntityKeyValues_GetVector2D);
DEFINE_NATIVE("CEntityKeyValues.GetVector4D", Bridge_CEntityKeyValues_GetVector4D);
DEFINE_NATIVE("CEntityKeyValues.GetQAngle", Bridge_CEntityKeyValues_GetQAngle);
DEFINE_NATIVE("CEntityKeyValues.SetBool", Bridge_CEntityKeyValues_SetBool);
DEFINE_NATIVE("CEntityKeyValues.SetInt", Bridge_CEntityKeyValues_SetInt);
DEFINE_NATIVE("CEntityKeyValues.SetUint", Bridge_CEntityKeyValues_SetUint);
DEFINE_NATIVE("CEntityKeyValues.SetInt64", Bridge_CEntityKeyValues_SetInt64);
DEFINE_NATIVE("CEntityKeyValues.SetUint64", Bridge_CEntityKeyValues_SetUint64);
DEFINE_NATIVE("CEntityKeyValues.SetFloat", Bridge_CEntityKeyValues_SetFloat);
DEFINE_NATIVE("CEntityKeyValues.SetDouble", Bridge_CEntityKeyValues_SetDouble);
DEFINE_NATIVE("CEntityKeyValues.SetString", Bridge_CEntityKeyValues_SetString);
DEFINE_NATIVE("CEntityKeyValues.SetPtr", Bridge_CEntityKeyValues_SetPtr);
DEFINE_NATIVE("CEntityKeyValues.SetStringToken", Bridge_CEntityKeyValues_SetStringToken);
DEFINE_NATIVE("CEntityKeyValues.SetColor", Bridge_CEntityKeyValues_SetColor);
DEFINE_NATIVE("CEntityKeyValues.SetVector", Bridge_CEntityKeyValues_SetVector);
DEFINE_NATIVE("CEntityKeyValues.SetVector2D", Bridge_CEntityKeyValues_SetVector2D);
DEFINE_NATIVE("CEntityKeyValues.SetVector4D", Bridge_CEntityKeyValues_SetVector4D);
DEFINE_NATIVE("CEntityKeyValues.SetQAngle", Bridge_CEntityKeyValues_SetQAngle);